src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -293,6 +295,68 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
+    * @param userInfo       UserInfo
+    * @param props          SpreadsheetProperties
+    * @return               JsObject representing the Google Create response
+    *
+    */
+  def createSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, props: SpreadsheetProperties)
+    (implicit actorRefFactory: ActorRefFactory, executionContext: ExecutionContext): Future[JsObject] = {
+
+    getUserProfile(requestContext) map { userResponse =>
+      val responseStr = userResponse.entity.toString
+      userResponse.status match {
+        case OK =>
+          val credential = new GoogleCredential().setAccessToken(userInfo.accessToken.token)
+          val service = new Sheets.Builder(httpTransport, jsonFactory, credential).setApplicationName("FireCloud").build()
+          val spreadsheet = new Spreadsheet().setProperties(props)
+          val response = service.spreadsheets().create(spreadsheet).execute()
+          val spreadsheetJson = response.toString.parseJson.asJsObject
+          logger.debug(s"Spreadsheet Create Response: ${spreadsheetJson.toString}")
+          spreadsheetJson
+        case x =>
+          log.error(s"Failed to save a google sheets service for [$responseStr]")
+          throw new FireCloudException(s"Failed to build a google sheets service for [$responseStr]")
+      }
+    }
+  }
+
+  /**
+    * Updates an existing google drive spreadsheet with the provided data.
+    *
+    * @param requestContext RequestContext
+    * @param userInfo       UserInfo
+    * @param spreadsheetId  Spreadsheet ID
+    * @param range          Range for where data updates begin, i.e. "Sheet1!A1" or "Sheet2!B1", etc.
+    * @param content        ValueRange
+    * @return               JsObject representing the Google Update response
+    *
+    */
+  def updateSpreadsheet(requestContext: RequestContext, userInfo: UserInfo, spreadsheetId: String, range: String, content: ValueRange)
+    (implicit actorRefFactory: ActorRefFactory, executionContext: ExecutionContext): Future[JsObject] = {
+    getUserProfile(requestContext) map { userResponse =>
+      val responseStr = userResponse.entity.toString
+      userResponse.status match {
+        case OK =>
+          val credential = new GoogleCredential().setAccessToken(userInfo.accessToken.token)
+          val service = new Sheets.Builder(httpTransport, jsonFactory, credential).setApplicationName("FireCloud").build()
+          val response = service.spreadsheets().values().update(spreadsheetId, range, content).setValueInputOption("RAW").execute()
+          val spreadsheetJson = response.toString.parseJson.asJsObject
+          logger.debug(s"Spreadsheet Update Response: ${spreadsheetJson.toString}")
+          spreadsheetJson
+        case x =>
+          log.error(s"Failed to save a google sheets service for [$responseStr]")
+          throw new FireCloudException(s"Failed to build a google sheets service for [$responseStr]")
+      }
+    }
+  }
```

Comment:
does this overwrite existing data in the spreadsheet? In other words, if some campaign manager manually adds comments to a cell or formatting or something ... will those be lost?

Reply:
It does overwrite the data, but does not change cell formatting. The way I envision this working would be that orch writes to the same worksheet columns every time, and all charts, additional columns (outside of what we update), etc. would remain in place. I think a best practice would be that the worksheet orch touches is never modified by users and instead, they make their changes in other worksheets.

Reply:
Additionally, it might make sense to write a set of instructions in one worksheet, data in another, and let the user do what they want in all of the others.